File: 2428-equal-row-and-column-pairs/equal-row-and-column-pairs.py
from collections import defaultdict
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        ret = 0
        rows = defaultdict(int)
        n = len(grid)
        for i in range(n):
            curr = tuple(grid[i])
            rows[curr] = rows.get(curr,0)+1
        for i in range(n):
            col = [grid[j][i] for j in range(n)]
            ret+= rows.get(tuple(col),0)
        return ret

File: 2428-equal-row-and-column-pairs/equal-row-and-column-pairs.cpp
#include <iostream>
#include <map>
#include <vector>
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        int ret = 0;
        map<vector<int>,int> lookup;
        int n = grid.size();
        ////////////////////////////
        for(int i = 0; i<n; i++){
            lookup[grid[i]]++;
        }
        ////////////////////////////
        for (int c = 0; c < n; c++){
            vector<int> col;
            for (int i = 0; i < n;i++){
                col.push_back(grid[i][c]);
            }
            ret += lookup[col];
        }
    return ret;
    }
};